std/datetime/test.ts

```diff
@@ -113,6 +113,19 @@ Deno.test({
   },
 });
 
+Deno.test({
+  name: "[std/datetime] weekOfMonth",
+  fn: () => {
+    assertEquals(datetime.weekOfMonth(new Date("2019-01-01T03:24:00")), 1);
+    assertEquals(datetime.weekOfMonth(new Date("2019-01-08T03:24:00")), 2);
+    assertEquals(datetime.weekOfMonth(new Date("2019-01-15T03:24:00")), 3);
+    assertEquals(datetime.weekOfMonth(new Date("2019-01-22T03:24:00")), 4);
+    assertEquals(datetime.weekOfMonth(new Date("2019-01-29T03:24:00")), 5);
+    assertEquals(datetime.weekOfMonth(new Date("2019-03-11T03:24:00")), 3);
+    assertEquals(datetime.weekOfMonth(new Date("2019-12-31T03:24:00")), 6);
```

Comment:
There were 6 weeks in December 2019? Not according to my calendar.

<img width="225" alt="Screen Shot 2020-08-17 at 5 44 32 PM" src="https://user-images.githubusercontent.com/80/90447456-5948d500-e0b1-11ea-85eb-d71186dfa257.png">

Reply:
A large part of the world (including ISO who this standard comes from) starts their weeks on Mondays :-)

Reply:
I don't get why we even need this function. It only makes sense if it doesn't count no partial weeks. But it does, which means that week 6 of dec 2019 is also week 1 of jan 2020 :-/

> The ISO standard does not define any association of weeks to months. A date is either expressed with a month and day-of-the-month, or with a week and day-of-the-week, never a mix.
> _From https://en.wikipedia.org/wiki/ISO_week_date#Weeks_per_month_

What is the use case for this function anyway @timreichen?

Reply:
It's quite specialized. I don't want to be reviewing a bunch of complex date-time computations - these things are notoriously difficult to get correct and I am not an expert.

@timreichen Thank you for the patch but I think this works better as an external module: https://deno.land/x#add

Closing without merge.

Reply:
@ry but as the deno manual says, std shouldn‘t have external deps. If we want to extend the symbol set for the datetime parser and formatter (symbols `q`, `qq` and `F`) we need these functions anyway. Or is there no intent of having a more complete formatter?
I think they are not more specialized than `weekOfYear` or `isLeap`.
In my opinion std is the right place for a basic set of such functions since languages like go, swift, rust python all have datetime parsing/formatting built-in.

Reply:
> Or is there no intent of having a more complete formatter?

We just don't have any datetime experts... 

> I think they are not more specialized than weekOfYear or isLeap.

Yes, I was worried already about these. I have no idea if they are correct.

Maybe if we could look up some test cases from Go, and port these over, so it would be easy to check....

Reply:
@ry 
> Yes, I was worried already about these. I have no idea if they are correct.
I think they are fine.
`isLeap ` seems to be a direct [port](https://golang.org/src/time/time.go) from go.
`dayOfYear` ported tests #7105